taipy/gui/utils/_map_dict.py

```diff
@@ -23,7 +24,7 @@ class _MapDict(object):
     __local_vars = ("_dict", "_update_var")
 
     def __init__(self, dict_import: dict, app_update_var=None):
-        self._dict = dict_import
+        self._dict = copy.deepcopy(dict_import)
```

Comment:
I'm afraid that will generate a *lot* of copying.
I guess we would need to add a static method to _MapDict for internal use that would *not* do the deepcopy and would prevent some undesirable side effects 
ex: _MapDict.__getitem__ (_MapDict.py:L36)